examples/vision/image_classification_efficientnet_fine_tuning.py

```diff
@@ -0,0 +1,423 @@
+(https://arxiv.org/abs/1805.09501).
+
+For B0 to B7 base models, the input shapes are different. Here is a list of input shape
+expected for each model:
+
+| Base model | resolution|
+|----------------|-----|
+| EfficientNetB0 | 224 |
+| EfficientNetB1 | 240 |
+| EfficientNetB2 | 260 |
+| EfficientNetB3 | 300 |
+| EfficientNetB4 | 380 |
+| EfficientNetB5 | 456 |
+| EfficientNetB6 | 528 |
+| EfficientNetB7 | 600 |
+
+When the use of the model is intended for transfer learning, the Keras implementation
+provides a option to remove the top layers:
+```
+model = EfficientNetB0(include_top=False, weights='imagenet')
+```
+This option excludes the final Dense layer that turns 1280 features on the penultimate
+layer into prediction of the 1000 classes in imagenet. Replacing the top with custom
+layers allows using EfficientNet as a feature extractor and transfers the pretrained
+weights to other tasks.
+
+Another keyword in the model builder worth noticing is `drop_connect_rate` which controls
+the dropout rate responsible for stochastic depth (https://arxiv.org/abs/1603.09382).
+This parameter serves as a toggle for extra regularization in finetuning, but does not
+alter loaded weights.
+
+
+
+## Example: EfficientNetB0 for CIFAR-100.
+
+As an architecture, EfficientNet is capable of a wide range of image classification
+tasks. For example, we will show using pre-trained EfficientNetB0 on CIFAR-100. For
+EfficientNetB0, image size is 224.
+"""
+
+# IMG_SIZE is determined by EfficientNet model choice
+IMG_SIZE = 224
+
+"""
+### prepare
+"""
+
+"""shell
+!pip install --quiet tensorflow==2.3.0rc0
+!pip install --quiet cloud-tpu-client
+"""
+import tensorflow as tf
+
+try:
+    from cloud_tpu_client import Client
+
+    c = Client()
+    c.configure_tpu_version(tf.__version__, restart_type="always")
+    tpu = tf.distribute.cluster_resolver.TPUClusterResolver()  # TPU detection
+    print("Running on TPU ", tpu.cluster_spec().as_dict()["worker"])
+    tf.config.experimental_connect_to_cluster(tpu)
+    tf.tpu.experimental.initialize_tpu_system(tpu)
+    strategy = tf.distribute.experimental.TPUStrategy(tpu)
+except ValueError:
+    print("Not connected to a TPU runtime. Using CPU/GPU strategy")
+    strategy = tf.distribute.MirroredStrategy()
+
+
+"""
+Below is example code for loading data.
+To see sensible result, you need to load entire dataset and adjust epochs for
+training; but you may truncate data for a quick verification of the workflow.
+Expect the notebook to run at least an hour for GPU, while much faster on TPU if
+using hosted Colab session.
+"""
+
+from tensorflow import keras
+from tensorflow.keras.datasets import cifar100
+from tensorflow.keras.utils import to_categorical
+
+batch_size = 64
+
+(x_train, y_train), (x_test, y_test) = cifar100.load_data()
+NUM_CLASSES = 100
+
+x_train = tf.cast(x_train, tf.int32)
+x_test = tf.cast(x_test, tf.int32)
+
+truncate_data = False  # @param {type: "boolean"}
```

Comment:
Just curious why we need this option, is it for testing with smaller amount of data? Probably add a comment for this.

Reply:
Yes this is for testing with smaller amount of data.  The comment makes an interactive button when opened with colab. 
Sure, I should add a comment to clearly explain.